Add Fedora multilib support to vglrun. Previously, it was only supported under
Solaris using LD_PRELOAD_32 and LD_PRELOAD_64. This can be done on Linux using
simple if statements in bash.

Some paths were also fixed in this patch.
diff -Nru VirtualGL-2.3.2.orig/server/vglrun VirtualGL-2.3.2/server/vglrun
--- VirtualGL-2.3.2.orig/server/vglrun	2012-10-12 15:38:39.798410146 -0400
+++ VirtualGL-2.3.2/server/vglrun	2012-10-12 16:45:30.445347958 -0400
@@ -1,4 +1,7 @@
-#!/bin/sh
+#!/bin/bash
+
+# Modifications for Fedora multilib support is:
+# Copyright (C)2012 Xiao-Long Chen <chenxiaolong@cxl.epac.to>
 
 # Copyright (C)2004 Landmark Graphics Corporation
 # Copyright (C)2005 Sun Microsystems, Inc.
@@ -106,8 +109,8 @@
 	echo
 	echo "+/-v      : Enable/disable verbose VirtualGL messages [default = disabled]"
 	echo
-	echo "-64       : (Solaris only) Preload VirtualGL only into 64-bit apps"
-	echo "-32       : (Solaris only) Preload VirtualGL only into 32-bit apps"
+	echo "-64       : Preload VirtualGL only into 64-bit apps"
+	echo "-32       : Preload VirtualGL only into 32-bit apps"
 	echo
 	exit $1
 }
@@ -183,15 +186,12 @@
 	usage 0
 fi
 
-if [ -r /etc/opt/VirtualGL/vgl_xauth_key ]; then
+if [ -r /etc/VirtualGL/vgl_xauth_key ]; then
 	XAUTH=xauth
-	if [ -x /usr/X11R6/bin/xauth ]; then
-		XAUTH=/usr/X11R6/bin/xauth
-	fi
-	if [ -x /usr/openwin/bin/xauth ]; then
-		XAUTH=/usr/openwin/bin/xauth
-	fi
-	$XAUTH merge /etc/opt/VirtualGL/vgl_xauth_key
+	if [ -x /usr/bin/xauth ]; then
+		XAUTH=/usr/bin/xauth
+	fi
+	$XAUTH merge /etc/VirtualGL/vgl_xauth_key
 fi
 
 SSH_IP=`echo $SSH_CLIENT | cut -d' ' -f1 | cut -d: -f4`
@@ -214,94 +214,63 @@
 	fi
 fi
 
-case `uname -s` in
-
-SunOS)
-
-if [ -z "$FAKERLIB" ]; then
-FAKERLIB=librrfaker.so
-if [ -f /opt/VirtualGL/lib/64/librrfaker.so ]
-then
-	FAKERLIB=/opt/VirtualGL/lib/64/librrfaker.so
-fi
-fi
-export FAKERLIB
-if [ -z "$FAKERLIB32" ]; then
-FAKERLIB32=librrfaker.so
-if [ -f /opt/VirtualGL/lib/librrfaker.so ]
-then
-	FAKERLIB32=/opt/VirtualGL/lib/librrfaker.so
-fi
-fi
-export FAKERLIB32
-if [ $__VGL_64 -eq 0 ]; then
-	if [ -z "$LD_PRELOAD_32" ]; then
-		LD_PRELOAD_32=$FAKERLIB32
+case $(uname -m) in
+i686)
+	# Fail if "-x64" is specified on a x86_32 system
+	if [ "x$(uname -m)" = "xi686" ] && [ "x${__VGL_64}" = "x1" ]; then
+		echo "Cannot preload 64-bit libraries on a 32-bit machine."
+		exit 1
+	fi
+	# Preload x86_32 libraries
+	if [ -z "$LD_PRELOAD" ]; then
+		LD_PRELOAD=/usr/lib/librrfaker.so
 	else
-		LD_PRELOAD_32=$FAKERLIB32:$LD_PRELOAD_32
+		LD_PRELOAD=/usr/lib/librrfaker.so:$LD_PRELOAD
+	fi
+	if [ $__VGL_DL -eq 1 ]; then
+		LD_PRELOAD=/usr/lib/libdlfaker.so:$LD_PRELOAD
 	fi
-	if [ $__VGL_DL -eq 1 ]; then
-		if [ -f /opt/VirtualGL/lib/libdlfaker.so ]
-		then
-			LD_PRELOAD_32=/opt/VirtualGL/lib/libdlfaker.so:$LD_PRELOAD_32
-		else
-			LD_PRELOAD_32=libdlfaker.so:$LD_PRELOAD_32
-		fi
+	if [ $__VGL_GE -eq 1 ]; then
+		LD_PRELOAD=/usr/lib/libgefaker.so:$LD_PRELOAD
 	fi
-	if [ $__VGL_GE -eq 1 ]; then
-		if [ -f /opt/VirtualGL/lib/libgefaker.so ]
-		then
-			LD_PRELOAD_32=/opt/VirtualGL/lib/libgefaker.so:$LD_PRELOAD_32
+	;;
+x86_64)
+	# Preload x86_32 libraries if "-32" is specified
+	if [ "x${__VGL_32}" = "x1" ]; then
+		# If the 32-bit multilib libVirtualGL is not installed, then
+		# alert the user
+		if [ ! -f /usr/lib/librrfaker.so ]; then
+			echo "Please install the i686 version of libVirtualGL to run 32-bit apps on a 64-bit system."
+			exit 1
+		fi
+		if [ -z "$LD_PRELOAD" ]; then
+			LD_PRELOAD=/usr/lib/librrfaker.so
 		else
-			LD_PRELOAD_32=libgefaker.so:$LD_PRELOAD_32
+			LD_PRELOAD=/usr/lib/librrfaker.so:$LD_PRELOAD
 		fi
-	fi
-	export LD_PRELOAD_32
-fi
-if [ $__VGL_32 -eq 0 ]; then
-	if [ -z "$LD_PRELOAD_64" ]; then
-		LD_PRELOAD_64=$FAKERLIB
+		if [ $__VGL_DL -eq 1 ]; then
+			LD_PRELOAD=/usr/lib/libdlfaker.so:$LD_PRELOAD
+		fi
+		if [ $__VGL_GE -eq 1 ]; then
+			LD_PRELOAD=/usr/lib/libgefaker.so:$LD_PRELOAD
+		fi
+
+	# Otherwise, preload x86_64 libraries
 	else
-		LD_PRELOAD_64=$FAKERLIB:$LD_PRELOAD_64
-	fi
-	if [ $__VGL_DL -eq 1 ]; then
-		if [ -f /opt/VirtualGL/lib/64/libdlfaker.so ]
-		then
-			LD_PRELOAD_64=/opt/VirtualGL/lib/64/libdlfaker.so:$LD_PRELOAD_64
+		if [ -z "$LD_PRELOAD" ]; then
+			LD_PRELOAD=/usr/lib64/librrfaker.so
 		else
-			LD_PRELOAD_64=libdlfaker.so:$LD_PRELOAD_64
+			LD_PRELOAD=/usr/lib64/librrfaker.so:$LD_PRELOAD
 		fi
-	fi
-	if [ $__VGL_GE -eq 1 ]; then
-		if [ -f /opt/VirtualGL/lib/64/libgefaker.so ]
-		then
-			LD_PRELOAD_64=/opt/VirtualGL/lib/64/libgefaker.so:$LD_PRELOAD_64
-		else
-			LD_PRELOAD_64=libgefaker.so:$LD_PRELOAD_64
+		if [ $__VGL_DL -eq 1 ]; then
+			LD_PRELOAD=/usr/lib64/libdlfaker.so:$LD_PRELOAD
+		fi
+		if [ $__VGL_GE -eq 1 ]; then
+			LD_PRELOAD=/usr/lib64/libgefaker.so:$LD_PRELOAD
 		fi
 	fi
-	export LD_PRELOAD_64
-fi
-exec ${1+"$@"}
-
-;;
-
-*)
+	;;
+esac
 
-if [ -z "$LD_PRELOAD" ]; then
-	LD_PRELOAD=librrfaker.so
-else
-	LD_PRELOAD=librrfaker.so:$LD_PRELOAD
-fi
-if [ $__VGL_DL -eq 1 ]; then
-	LD_PRELOAD=libdlfaker.so:$LD_PRELOAD
-fi
-if [ $__VGL_GE -eq 1 ]; then
-	LD_PRELOAD=libgefaker.so:$LD_PRELOAD
-fi
 export LD_PRELOAD
 exec ${1+"$@"}
-
-;;
-
-esac
